crates/red_knot_python_semantic/resources/mdtest/type_properties/is_subtype_of.md

```diff
@@ -0,0 +1,232 @@
+static_assert(not is_subtype_of(int, Literal[1]))
+
+static_assert(not is_subtype_of(Unknown, Unknown))
+static_assert(not is_subtype_of(Unknown, Literal[1]))
+static_assert(not is_subtype_of(Literal[1], Unknown))
+
+static_assert(not is_subtype_of(Any, Any))
+static_assert(not is_subtype_of(Any, Literal[1]))
+static_assert(not is_subtype_of(Literal[1], Any))
+
+static_assert(not is_subtype_of(Literal[1], str))
+static_assert(not is_subtype_of(Literal[1], Unknown | str))
+
+static_assert(not is_subtype_of(Literal[1, 2], Literal[1]))
+static_assert(not is_subtype_of(Literal[1, 2], Literal[1, 3]))
+```
+
+## `Never`
+
+```py
+from typing_extensions import Literal, Never
+from knot_extensions import is_subtype_of, static_assert
+
+static_assert(is_subtype_of(Never, Literal[1]))
+```
+
+## Literal types
+
+```py
+from typing_extensions import Literal, LiteralString
+from knot_extensions import is_subtype_of, static_assert
+
+static_assert(is_subtype_of(Literal[1], int))
+static_assert(is_subtype_of(Literal[1], object))
+
+static_assert(is_subtype_of(Literal[True], bool))
+static_assert(is_subtype_of(Literal[True], int))
+static_assert(is_subtype_of(Literal[True], object))
+
+static_assert(is_subtype_of(Literal["foo"], LiteralString))
+static_assert(is_subtype_of(Literal["foo"], str))
+static_assert(is_subtype_of(Literal["foo"], object))
+
+static_assert(is_subtype_of(LiteralString, str))
+static_assert(is_subtype_of(LiteralString, object))
+
+static_assert(is_subtype_of(Literal[b"foo"], bytes))
+static_assert(is_subtype_of(Literal[b"foo"], object))
+```
+
+## Tuple types
+
+```py
+from typing_extensions import Literal
+from knot_extensions import is_subtype_of, static_assert
+
+static_assert(is_subtype_of(tuple[()], tuple[()]))
+static_assert(is_subtype_of(tuple[Literal[42]], tuple[int]))
+static_assert(is_subtype_of(tuple[Literal[42], Literal["foo"]], tuple[int, str]))
+static_assert(is_subtype_of(tuple[int, Literal["foo"]], tuple[int, str]))
+static_assert(is_subtype_of(tuple[Literal[42], str], tuple[int, str]))
+
+static_assert(not is_subtype_of(tuple[()], tuple[Literal[1]]))
+static_assert(not is_subtype_of(tuple[Literal[42]], tuple[str]))
+
+static_assert(not is_subtype_of(tuple[tuple[int, ...]], tuple[Literal[2]]))
+static_assert(not is_subtype_of(tuple[Literal[2]], tuple[tuple[int, ...]]))
+```
+
+## Union types
+
+```py
+from typing_extensions import Literal
+from knot_extensions import is_subtype_of, static_assert
+
+static_assert(is_subtype_of(Literal[1], int | str))
+static_assert(is_subtype_of(str | int, object))
+static_assert(is_subtype_of(Literal[1, 2], Literal[1, 2, 3]))
+```
+
+## Intersection types
+
+```py
+from typing_extensions import Literal, LiteralString
+from knot_extensions import Intersection, Not, is_subtype_of, static_assert
+
+static_assert(is_subtype_of(Intersection[int, Not[Literal[2]]], int))
+static_assert(is_subtype_of(Intersection[int, Not[Literal[2]]], Intersection[Not[Literal[2]]]))
```

Comment:
`Intersection[Not[T]]` is equivalent to just `Not[T]`:

```suggestion
static_assert(is_subtype_of(Intersection[int, Not[Literal[2]]], Not[Literal[2]]))
```

The same simplification applies to a few other tests below as well.